### ISSUE TITLE
Add a sparse-checkout option to git repository rules

### ISSUE THREAD

**Issue by anthonymahshigian** [bazelbuild/bazel#24069]
### Description of the feature request:

Hi there! I'm requesting that git_override (in bzlmod) be expanded to add a sparse-checkout option. This could be done relatively easy with a custom repository rule back with Workspace, but now with bzlmod I'm hoping to not have to use a module extension to achieve this.
https://bazel.build/rules/lib/globals/module#git_override

### Which category does this issue belong to?

External Dependency

### What underlying problem are you trying to solve with this feature?

I am dealing with a very large repository, and using the current git_override rule is not feasible.

### Which operating system are you running Bazel on?

n/a

### What is the output of `bazel info release`?

n/a

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.

n/a

### What's the output of `git remote get-url origin; git rev-parse HEAD` ?

```text
n/a
```

### Have you found anything relevant by searching the web?

I found https://github.com/bazelbuild/bazel/issues/13747 that requests a similar feature for git_repository.

### Any other information, logs, or outputs that you want to share?

I don't believe this would be too difficult. The additional inputs would be a list of repository paths to checkout (passed down to git). We would also have to enforce the system's git is a minimum version. I'd be happy to take a stab at it myself if someone could confirm this as a valuable addition and point me to the code to modify.

**Comment by meteorcloudy** [comment]
@lberki Do you think this is doable only for git_repository?

**Comment by lberki** [comment]
I'm not entirely sure what "only for `git_repository`" means, but it looks like `git_worker.bzl` calls out to the `git` command line tool, so it should be implementable in `git_repository()` and `git_override` uses that, so I think the answer to you question is "yes"?

**Comment by meteorcloudy** [comment]
Yes, the scope is much smaller than adding native support of git sparse-checkout in Bazel. So contribution is welcome!

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 8.2.0 RC3](https://github.com/bazelbuild/bazel/releases/tag/8.2.0rc3). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=8.2.0rc3. Thanks!
